components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1251,30 +1251,66 @@ where
                 // already is a force leader, do nothing
                 return;
             }
+            Some(ForceLeaderState::WaitTicks { .. }) => {
+                self.fsm.peer.force_leader = None;
+            }
             None => {}
         }
 
-        if self.fsm.peer.is_leader() {
-            warn!(
-                "reject pre force leader due to already being leader";
-                "region_id" => self.fsm.region_id(),
-                "peer_id" => self.fsm.peer_id(),
-            );
-            return;
-        }
-
+        let ticks = if self.fsm.peer.is_leader() {
+            if self.fsm.hibernate_state.group_state() == GroupState::Ordered {
+                warn!(
+                    "reject pre force leader due to already being leader";
+                    "region_id" => self.fsm.region_id(),
+                    "peer_id" => self.fsm.peer_id(),
+                );
+                return;
+            }
+            // wait two rounds of election timeout to trigger check quorum to step down the leader
```

Comment:
It should check quorum the very time elapsed is equal to timeout.

Reply:
Need to trigger `CheckQuorum` twice, the first round is to set `recent_active = false`, the second round is to step down the leader. The leader may not step down within one round, due to `recent_active` may be true